timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -272,7 +272,7 @@ export class TimeGraphChart extends TimeGraphChartLayer {
                 this.mouseEndX = e.offsetX;
                 const start = this.mouseZoomingStart;
                 const end = this.unitController.viewRange.start + (this.mouseEndX / this.stateController.zoomFactor);
-                if (start !== end) {
+                if (Math.trunc(start) !== Math.trunc(end)) {
```

Comment:
I think this could happen if the range is too close but on both sides of an integer. I would suggest to use Math.round() instead.

However, if the current range is > 1 ns, and we zoom into this condition, I don't think we should ignore it, I think that instead we should make it so that if rounded start and end are the same, set end = start + 1. So for the user it would still zoom but to a 1 ns window range.

Also, I found that the freezing is due to time-graph-axis-scale.ts, at line 94, because the stepLength is so small, iLo and iHi are very large numbers, beyond integer number precision and in the for loop, the i value is unaffected by the i++ so that loop is infinite.